src/App.tsx

```diff
@@ -15,36 +16,84 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  function sortGoodsList(goods: string[], sortBy: string, isReverse: boolean) {
+    const sortedList = [...goods].sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+
+    return isReverse ? sortedList.reverse() : sortedList;
```

Comment:
[CODE KNOWLEDGE] - Don't interact with DOM directly, use React as much as possible. While this function itself does not interact with the DOM, it's important to ensure that state updates that result in DOM changes are handled through React state mechanisms. Consider moving the sorting logic into a useEffect hook that runs when `sortType` or `reversed` changes, and store the sorted goods in a state variable.